packages/bitcore-node/src/modules/ripple/models/transaction.ts

```diff
@@ -0,0 +1,246 @@
+import * as _ from 'lodash';
+import { ObjectID } from 'bson';
+import logger from '../../../logger';
+import { LoggifyClass } from '../../../decorators/Loggify';
+import { StorageService, Storage } from '../../../services/storage';
+import { partition } from '../../../utils/partition';
+import { TransformOptions } from '../../../types/TransformOptions';
+import { MongoBound } from '../../../models/base';
+import { SpentHeightIndicators } from '../../../types/Coin';
+import { EventStorage } from '../../../models/events';
+import { Config } from '../../../services/config';
+import { StreamingFindOptions } from '../../../types/Query';
+import { BaseTransaction } from '../../../models/baseTransaction';
+import { valueOrDefault } from '../../../utils/check';
+import { IXrpTransaction, XrpTransactionJSON, IXrpCoin } from '../types';
+import { CoinStorage } from '../../../models/coin';
+
+@LoggifyClass
+export class XrpTransactionModel extends BaseTransaction<IXrpTransaction> {
+  constructor(storage: StorageService = Storage) {
+    super(storage);
+  }
+
+  onConnect() {
+    super.onConnect();
+    this.collection.createIndex({ chain: 1, network: 1, from: 1, nonce: 1 }, { background: true, sparse: true });
```

Comment:
is this meant to be there globally? already on eth model

Reply:
I think it should be there, in case you never enabled the ETH one. There's nothing bad that can happen from calling this twice right?